x/mongo/driver/operation.go

```diff
@@ -433,10 +433,11 @@ func (op Operation) Execute(ctx context.Context) error {
 		// size to 16MiB because that's the maximum wire message size supported by MongoDB.
 		//
 		// Comment copied from https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/fmt/print.go;l=147
-		if cap(*wm) > 16*1024*1024 {
-			return
+		//
+		// Recycle byte slices that are smaller than 16MiB and at least half occupied.
+		if c := cap(*wm); c < 16*1024*1024 && c/2 < len(*wm) {
+			memoryPool.Put(wm)
 		}
```

Comment:
Actually, if I recycle the low-occupied slices as well, the benchmark shows an even lower memory consumption.